test/jdk/java/lang/invoke/lambda/TestStableLambdaName.java

```diff
@@ -0,0 +1,301 @@
+ * @summary Test if the names of the lambda classes are stable when -Djdk.internal.lambda.stableLambdaName
+ *          flag is set to true. This test directly calls java.lang.invoke.LambdaMetafactory#altMetafactory
+ *          method to create multilple lambda instances and then checks their names stability. We created a
+ *          multidimensional space of possible values for each parameter that java.lang.invoke.LambdaMetafactory#altMetafactory
+ *          takes and then search that space by combining different values of those parameters. There is a rule we have to follow:
+ *              - Alternative methods of the specific method must have the same signature with difference in parameter types
+ *                as long as the parameter of the alternative method is the superclass type of the type of corresponding parameter in
+ *                original method
+ * @run main/othervm -Djdk.internal.lambda.stableLambdaName=true TestStableLambdaName
+ */
+
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.rmi.Remote;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+public class TestStableLambdaName {
+    private static final MethodHandles.Lookup lookup = MethodHandles.lookup();
+
+    // Different types of lambda classes based on value of flags parameter in the
+    // java.lang.invoke.LambdaMetafactory#altMetafactory.
+    // java.lang.invoke.LambdaMetafactory#altMetafactory uses bitwise and with this
+    // parameter and predefined values to determine if lambda is serializable, has
+    // altMethods and altInterfaces etc.
+    private enum lambdaType {
+        NOT_SERIALIZABLE_NO_ALT_METHODS_NO_ALT_INTERFACES (0),
+        SERIALIZABLE_ONLY (1),
+        NOT_SERIALIZABLE_HAS_ALT_INTERFACES(2),
+        SERIALIZABLE_HAS_ALT_INTERFACES(3),
+        NOT_SERIALIZABLE_HAS_ALT_METHODS(4),
+        SERIALIZABLE_HAS_ALT_METHODS(5),
+        NOT_SERIALIZABLE_HAS_ALT_METHODS_HAS_ALT_INTERFACES(6),
+        SERIALIZABLE_HAS_ALT_METHODS_HAS_ALT_INTERFACES(7);
+
+        private final int index;
+        lambdaType(int i) {
+            index = i;
+        }
+    }
+
+    private static final String[] interfaceMethods = new String[]{"accept", "consume", "apply", "supply", "get", "test", "getAsBoolean"};
+    private static final Class<?>[] interfaces = new Class<?>[]{Consumer.class, Function.class, Predicate.class, Supplier.class, BooleanSupplier.class};
+    // List of method types for defined methods
+    private static final MethodType[] methodTypes = new MethodType[]{MethodType.methodType(String.class, Integer.class), MethodType.methodType(Throwable.class, AssertionError.class)};
+    private static MethodHandle[] methodHandles;
```

Comment:
This is also strange to be static.

Reply:
Same as above. Maybe, solution is to create an instance of this class in `main` and then just to call it's method. Then, we would not need any of these class fields to be static.